scheduler/src/cook/progress.clj

```diff
@@ -57,13 +59,16 @@
         (let [old-count (count instance-id->progress-state)
               new-count (count instance-id->progress-state')]
           (when (zero? old-count)
+            (prom/set prom/progress-aggregator-pending-states-count 0)
             (counters/clear! progress-aggregator-pending-states-count))
+          (prom/inc prom/progress-aggregator-pending-states-count (- new-count old-count))
           (counters/inc! progress-aggregator-pending-states-count (- new-count old-count)))
         instance-id->progress-state')
       (do
         (log/warn "skipping" data "as it is missing an integer progress-sequence")
         instance-id->progress-state))
     (do
+      (prom/inc prom/progress-aggregator-drop-count)
```

Comment:
Does this replace both the rate and the count?

Reply:
Yes, we can use the counter variable with a `rate` function to calculate the rate dynamically.